sdk/core/Azure.Core/src/Shared/OperationInternal.cs

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class OperationInternal<TResult>
+    {
+        private readonly TimeSpan _defaultPollingInterval;
+
+        private readonly IOperation<TResult> _operation;
+
+        private TResult _value;
+
+        public OperationInternal(string operationTypeName,
+            ClientDiagnostics clientDiagnostics,
+            IOperation<TResult> operation,
+            TimeSpan? defaultPollingInterval = null) : this(operationTypeName, clientDiagnostics, defaultPollingInterval)
```

Comment:
Making the default polling interval customizable in the constructor since some clients use different values. We can also remove it from the constructor and make it a settable or initializable property. I left it in the constructor for visibility, but we may consider moving it given that not many clients use custom values.